[en]: Update insight probe method docs (#5605)
DaoCloud/DaoCloud-docs: docs/en/docs/insight/user-guide/collection-manag/probe-module.md

---
MTPE: WANG0608GitHub
Date: 2024-09-23
---

> # Custom Probe Methods

Reviewer: ```suggestion
# Custom Probes
```

Insight uses the Blackbox Exporter provided by Prometheus as a blackbox monitoring solution, allowing detection of target instances via HTTP, HTTPS, DNS, ICMP, TCP, and gRPC. It can be used in the following scenarios:

- HTTP/HTTPS: URL/API availability monitoring
- ICMP: Host availability monitoring
- TCP: Port availability monitoring
- DNS: Domain name resolution

> In this page, we will explain how to configure custom probe methods in an existing Blackbox ConfigMap.

Reviewer: ```suggestion
In this page, we will explain how to configure custom probes in an existing Blackbox ConfigMap.
```

> ICMP probing is not enabled by default in Insight because it requires higher permissions. Therfore We will use the HTTP probe method as an example to demonstrate how to modify the ConfigMap to achieve custom HTTP probing.

Reviewer: ```suggestion
ICMP probing is not enabled by default in Insight because it requires higher permissions. Therfore We will use the HTTP probe as an example to demonstrate how to modify the ConfigMap to achieve custom HTTP probing.
```

## Procedure

1. Go to __Clusters__ in __Container Management__ and enter the details of the target cluster.
2. Click the left navigation bar and select __ConfigMaps & Secrets__ -> __ConfigMaps__ .
3. Find the ConfigMap named __insight-agent-prometheus-blackbox-exporter__ and click __Edit YAML__ .

>     Add custom probe methods under __modules__ :

Reviewer: ```suggestion
    Add custom probes under __modules__ :
```

> === "HTTP Probing"

Reviewer: ```suggestion
=== "HTTP Probe"
```

          method: GET
    ```

> === "ICMP Probing"

Reviewer: ```suggestion
=== "ICMP Probe"
```

    ```yaml
    module:
      ICMP: # Example of ICMP probing configuration
        prober: icmp
        timeout: 5s
        icmp:
          preferred_ip_protocol: ip4
    icmp_example: # Example 2 of ICMP probing configuration
      prober: icmp
      timeout: 5s
      icmp:
        preferred_ip_protocol: "ip4"
        source_ip_address: "127.0.0.1"
    ```
    Since ICMP requires higher permissions, we also need to elevate the pod permissions. Otherwise, an `operation not permitted` error will occur. There are two ways to elevate permissions:

>     1. Method One: Directly edit the `BlackBox Exporter` deployment file to enable it

Reviewer: ```suggestion
    - Directly edit the `BlackBox Exporter` deployment file to enable it
```
中文可以同步修改，不是1-2，是并列的

       ```yaml
        apiVersion: apps/v1
        kind: Deployment
        metadata:
          name: insight-agent-prometheus-blackbox-exporter
          namespace: insight-system
        spec:
          template:
            spec:
              containers:
                - name: blackbox-exporter
                  image: # ... (image, args, ports, etc. remain unchanged)
                  imagePullPolicy: IfNotPresent
                  securityContext:
                    allowPrivilegeEscalation: false
                    capabilities:
                      add:
                      - NET_RAW
                      drop:
                      - ALL
                    readOnlyRootFilesystem: true
                    runAsGroup: 0
                    runAsNonRoot: false
                    runAsUser: 0
       ```

>     2. Method Two: Elevate permissions via Helm Upgrade

Reviewer: ```suggestion
    - Elevate permissions via `helm upgrade`
```

       ```diff
        prometheus-blackbox-exporter:
          enabled: true
          securityContext:
            runAsUser: 0
            runAsGroup: 0
            readOnlyRootFilesystem: true
            runAsNonRoot: false
            allowPrivilegeEscalation: false
            capabilities:
              add: ["NET_RAW"]
       ```

!!! info

>     For more probe methods, refer to [blackbox_exporter Configuration](https://github.com/prometheus/blackbox_exporter/blob/master/CONFIGURATION.md).

Reviewer: ```suggestion
    For more probes, refer to [blackbox_exporter Configuration](https://github.com/prometheus/blackbox_exporter/blob/master/CONFIGURATION.md).
```

          method: GET
    ```

=== "ICMP Prober"

    ```yaml
    module:
      ICMP: # Example of ICMP prober configuration
        prober: icmp
        timeout: 5s
        icmp:
          preferred_ip_protocol: ip4
    icmp_example: # Example 2 of ICMP prober configuration
      prober: icmp
      timeout: 5s
      icmp:
        preferred_ip_protocol: "ip4"
        source_ip_address: "127.0.0.1"
    ```
    Since ICMP requires higher permissions, we also need to elevate the pod permissions. Otherwise, an `operation not permitted` error will occur. There are two ways to elevate permissions:

    - Directly edit the `BlackBox Exporter` deployment file to enable it

>       ```yaml

Reviewer: ```suggestion
        ```yaml
```

        apiVersion: apps/v1
        kind: Deployment
        metadata:
          name: insight-agent-prometheus-blackbox-exporter
          namespace: insight-system
        spec:
          template:
            spec:
              containers:
                - name: blackbox-exporter
                  image: # ... (image, args, ports, etc. remain unchanged)
                  imagePullPolicy: IfNotPresent
                  securityContext:
                    allowPrivilegeEscalation: false
                    capabilities:
                      add:
                      - NET_RAW
                      drop:
                      - ALL
                    readOnlyRootFilesystem: true
                    runAsGroup: 0
                    runAsNonRoot: false
                    runAsUser: 0

>       ```

Reviewer: ```suggestion
        ```
```

    - Elevate permissions via `helm upgrade`

>       ```diff

Reviewer: ```suggestion
        ```diff
```

        prometheus-blackbox-exporter:
          enabled: true
          securityContext:
            runAsUser: 0
            runAsGroup: 0
            readOnlyRootFilesystem: true
            runAsNonRoot: false
            allowPrivilegeEscalation: false
            capabilities:
              add: ["NET_RAW"]

>       ```

Reviewer: ```suggestion
        ```
```
